app/controllers/products_controller.rb

```diff
@@ -10,8 +12,23 @@ def index
   def show
     @product = Product.find(params[:id])
     @stock_adjustments = @product.stock_adjustments.page(params[:page]).per(10).order(created_at: :desc)
+    qrcode = RQRCode::QRCode.new(product_url)
+
+    @png = qrcode.as_png(
+      bit_depth: 1,
+      border_modules: 4,
+      color_mode: ChunkyPNG::COLOR_GRAYSCALE,
+      color: "black",
+      file: nil,
+      fill: "white",
+      module_px_size: 6,
+      resize_exactly_to: false,
+      resize_gte_to: false,
+      size: 250
+    )
```

Comment:
I think this code should be moved from here 🤔 It exposes a lot of details that the controller should not be interested in. We can hide those details by moving this code to its own class. E.g. we could add a class `GetQRCode` in `app/queries` and call it from the controller: `@png = GetQRCode.call(product_url, size: 250, format: :png)`